src/components/tic-tac-toe.test.tsx

```diff
@@ -0,0 +1,278 @@
+import 'jsdom-global/register'; // at the top of file , even  , before importing react
+import * as React from 'react';
+import { render, mount } from 'enzyme';
+// import { JSDOM } from 'jsdom';
+// const doc = JSDOM.jsdom('<!doctype html><html><body></body></html>'); // Needed for mounting
+
+const TicTacToe = require('./tic-tac-toe');
+
+describe('Tic-Tac-Toe module', () => {
+  it('Expects baord to have 3 row children', () => {
+    const onClick = (i: number) => i + 1;
+    const board = render(
+      <TicTacToe.Board
+        squares={Array(9).fill(null)}
+        onClick={onClick}
+        winnerSquares={null}
+      />,
+    );
+    expect(board.children()).toHaveLength(3);
+  });
+
+  it('Expects each row to have 3 children', () => {
+    const row = render(
+      <TicTacToe.Row
+        rowNum={1}
+        squares={Array(3).fill(null)}
+        winnerSquares={null}
+        onClick={(i: number) => i + 1}
+      />,
+    );
+    expect(row.children()).toHaveLength(3);
+  });
+
+  it('Expects game to have the two components', () => {
+    const game = render(<TicTacToe.Game />);
+    expect(game.children()).toHaveLength(2);
+  });
+
+  it('expects board to render the state properly', () => {
+    const onClick = (i: number) => i + 1;
+    const board = mount(
+      <TicTacToe.Board
+        squares={['X', 'O', 'X', null, null, 'O', 'O', null, 'X']}
+        onClick={onClick}
+        winnerSquares={null}
+      />,
+    );
+    expect(
+      board

```

Comment:
You should *definitely* make a function for getting nested children from the board; writing `childAt(x).childAt(y).childAt(z)` over and over should immediately make you reconsider your approach.